lib/util/pssh.js

```diff
@@ -46,14 +47,27 @@ shaka.util.Pssh = class {
     this.cencKeyIds = [];
 
     /*
-    * Array of tuples that define the startIndex + size for each
-    * discrete pssh within |psshBox|
-    * */
+     * Array of tuples that define the startIndex + size for each
+     * discrete pssh within |psshBox|
+     */
     this.dataBoundaries = [];
 
-    new shaka.util.Mp4Parser()
-        .fullBox('pssh', (box) => this.parseBox_(box))
-        .parse(psshBox);
+    /**
+     * Array with the pssh founded.
+     * @type {!Array.<Uint8Array>}
+     */
+    this.data = [];
+
+    if (extractFromMoov) {
+      new shaka.util.Mp4Parser()
+          .box('moov', shaka.util.Mp4Parser.children)
+          .fullBox('pssh', (box) => this.parsePsshBox_(box))
+          .parse(psshBox);
+    } else {
+      new shaka.util.Mp4Parser()
+          .fullBox('pssh', (box) => this.parsePsshBox_(box))
+          .parse(psshBox);
+    }
```

Comment:
Based on my reading of Mp4Parser, I think you can just define the 'moov' box every time, whether it's present in the data buffer or not.

Reply:
I'll test it tomorrow. If it is true we can remove the if

Reply:
It's truth, I'm going to simplify it.